telethon/events/newmessage.py

```diff
@@ -200,12 +204,12 @@ class Event(EventCommon):
                 ...
                 >>>
         """
-        def __init__(self, message):
+        def __init__(self, message: types.TypeMessage):
             self.__dict__['_init'] = False
             super().__init__(chat_peer=message.peer_id,
                              msg_id=message.id, broadcast=bool(message.post))
 
-            self.pattern_match = None
+            self.pattern_match = None  # type: Optional[Pattern]
```

Comment:
This will be a `re.Match`, not `re.Pattern`. It can also actually be the type of whatever the `func=` returns, but that might get tricky.

Reply:
Well I didn't even know the `pattern_match` is actually the return value of `func`! I guess it's undocumented (or buried deeply in the docs). Yeah the Pattern should be Match.

Reply:
But I don't think mentioning the return type of the func is even possible, at least taking into consideration that we should support python 3.5. Because the type of the `func` should be dynamically fetched from it and get hinted to the variable.